src/main/java/net/minecraftforge/common/IAnvilRecipe.java

```diff
@@ -0,0 +1,64 @@
+package net.minecraftforge.common;
+
+import java.util.ArrayList;
+import net.minecraft.item.ItemStack;
+import net.minecraft.world.World;
+
+public interface IAnvilRecipe
+{
+    /**
+     * Returns the input for this recipe, any mod accessing this value should never
+     * manipulate the values in this array as it will effect the recipe itself.
+     *
+     * @return The recipes input vales. Each value can be an ItemStack, List<ItemStack> or null.
+     */
+    ArrayList<Object> getInput();
```

Comment:
I originally changed it to match the one in the crafting table recipes, but all that did was add complexity. I have a very similar recipe handler for one of my blocks, and after applying the same change to it I decided to change it back. Everything needed casting and checking before use, making it a complete pain.